core/shared/src/test/scala/fs2/TextSpec.scala

```diff
@@ -230,5 +234,28 @@ class TextSpec extends Fs2Spec {
         }
       }
     }
+
+    "base64Encode" in {
+      forAll { (bs: List[Array[Byte]]) =>
+        bs.map(Chunk.bytes).foldMap(Stream.chunk).through(text.base64Encode).compile.string shouldBe
+          bs.map(ByteVector.view(_)).foldLeft(ByteVector.empty)(_ ++ _).toBase64
+      }
+    }
+
+    "base64Encode andThen base64Decode" in {
+      forAll { (bs: List[Array[Byte]], unchunked: Boolean, rechunkSeed: Long) =>
+        bs.map(Chunk.bytes)
+          .foldMap(Stream.chunk)
+          .through(text.base64Encode)
+          .through {
+            if (unchunked) _.unchunk
+            else _.rechunkRandomlyWithSeed(0.1, 2.0)(rechunkSeed)
+          }
+          .through(text.base64Decode[Fallible])
+          .compile
+          .to(ByteVector) shouldBe
+          Right(bs.map(ByteVector.view(_)).foldLeft(ByteVector.empty)(_ ++ _))
+      }
+    }
```

Comment:
Could also add test feeding the decode invalid base64 characters and assert it fails as expected (not throws exception).

Reply:
Or invalid lengths.